SCClassLibrary/Common/Control/ReadableNodeIDAllocator.sc

```diff
@@ -0,0 +1,72 @@
+/* make nodeIDs more humanly readable:
+10 ** 8 ids for 1 user, 10 ** 7 for < 21, 10 ** 6 for < 210
+
+for example, with 20 Clients, clientIDs would be
+clientID  defaultGroup  tempIDs
+ 1         10000001      120001000, 10001001, etc
+ 2         10000001      120001000, 10001001, etc
+12        120000001      120001000, 10001001, etc
+
+In multi-client settings, this makes s.plotTree display
+or s.queryAllNodes posts much easier to understand.
+*/
+
+ReadableNodeIDAllocator {
+
+	var <clientID, <lowestTempID, <numClients;
+	var <numIDs, <idOffset, <maxPermID;
+	var tempCount = -1, permCount = 1, permFreed;
+
+	*new { arg clientID = 0, lowestTempID = 1000, numClients = 32;
+		^super.newCopyArgs(clientID, lowestTempID, numClients).reset
+	}
+
+	reset {
+		var maxNumPerUser = (2 ** 31 / numClients).trunc;
+		// go to next lower power of 10:
+		numIDs = (10 ** maxNumPerUser.log10.trunc).asInteger;
+
+		idOffset = numIDs * clientID;
+
+		permFreed = IdentitySet.new;
+		maxPermID = idOffset + lowestTempID - 1;
+		tempCount = -1;
+		permCount = 1;
+	}
+
+	alloc {
+		tempCount = tempCount + 1;
+		// wraps after handing out (numIDs - lowestTempID) ids
+		^(lowestTempID + tempCount).wrap(lowestTempID, numIDs) + idOffset;
+	}
+
+	isPerm { |num|
+		// 0 and 1 are also permanent
```

Comment:
sorry if I'm being dense.. i just don't understand what that has to do with the check below. was the point of this comment to say that we check `inclusivelyBetween` (as opposed to `exclusivelyBetween`) because idOffset + 0 should be a permanode if there's just one permanode?